sandbox-server/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+)
+
+func main() {
+	http.HandleFunc("/", handler)
+	http.ListenAndServe(":8080", nil)
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	user_id := r.URL.Query().Get("user_id")
+	algo_id := r.URL.Query().Get("algo_id")
+	if user_id == "" || algo_id == ""{
+		http.Error(w, "Missing required query parameter", http.StatusBadRequest)
+		return
+	}
+
+	ctx := context.Background()
```

Comment:
Use the request context (r.Context()) for operations that should be canceled if the HTTP request is canceled. Replace context.Background() with r.Context()